File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/base/RenderGroup.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.base;

import webgl.*;

/**
 * @author Norihiro
 *
 */
public class RenderGroup {
	public Material material;
	public Bone[] boneArray;
	public int[] indices;
	public WebGLBuffer indexBuffer;
	private int _indexDataCache;
	
	public RenderGroup(){
		
	}
	public DHVector4 getAmbient(){
		return null;
	}
	public void getBoneData(){
		
	}
	public DHVector4 getDiffuse(){
		return null;
	}
	public WebGLShortArray getIndexData(){
		return null;
	}
	public float getShininess(){
		return 0f;
	}
	public DHVector4 getSpecular(){
		return null;
	}
}

File: src/js/base/RenderGroup.js
/*--------------------------------------------------------------------------------
 * DH3DLibrary RenderGroup.js v0.2.0
 * Copyright (c) 2010-2012 DarkHorse
 *
 * DH3DLibrary is freely distributable under the terms of an MIT-style license.
 * For details, see the DH3DLibrary web site: http://darkhorse2.0spec.jp/dh3d/
 *
 *------------------------------------------------------------------------------*/
var RenderGroup = Class.create({
  material: null,
  boneArray: null,
  indices: null,
  indexBuffer: null,

  _indexDataCache: null,
  
  initialize: function() {
    this.indices = $A();
  },

  getBoneData: function() {
  },

  getIndexData: function() {
    if(!this._indexDataCache){
      this._indexDataCache = new Uint16Array(this.indices);
    }
    return this._indexDataCache;
  },

  getAmbient: function() {
    return this.material.getAmbient();
  },

  getDiffuse: function() {
    return this.material.getDiffuse();
  },

  getSpecular: function() {
    return this.material.getSpecular();
  },

  getShininess: function() {
    return this.material.getShininess();
  },

});
